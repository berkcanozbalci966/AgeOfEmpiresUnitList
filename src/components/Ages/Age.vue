<template>
  <div>
    <span> {{ ageInformation }} </span>
  </div>
</template>

<script>
export default {
  name: "Age",
  props: {
    ageInformation: {
      type: String,
      default: () => "",
    },
  },
};
</script>

<style lang="scss" scoped>
div {
  border: 1px solid black;
  padding: 4px;
  &:hover {
    cursor: pointer;
  }
}
</style>
