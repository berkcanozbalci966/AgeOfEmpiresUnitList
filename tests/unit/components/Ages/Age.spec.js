import { shallowMount } from "@vue/test-utils";
import Vuex from "vuex";
import Age from "../../../../src/components/Ages/Age";
import AgeModule from "../../../../src/store/modules/age";

describe("Age", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        Age: AgeModule,
      },
    });
  });

  it("Render correctly", () => {
    const component = shallowMount(Age, {
      global: {
        plugins: [store],
      },
      props: {
        ageInformation: {
          ageName: "Test",
        },
      },
    });

    expect(component.html()).toMatchSnapshot();
  });

  it("Selected class correctly", () => {
    const component = shallowMount(Age, {
      global: {
        plugins: [store],
      },
      props: {
        ageInformation: {
          ageName: "Test",
          selectedAge: "Test",
        },
      },
    });

    expect(component.classes("active")).toBe(true);
  });
  it("Click & emit", async () => {
    const component = shallowMount(Age, {
      global: {
        plugins: [store],
      },
      props: {
        ageInformation: {
          ageName: "Test",
          selectedAge: "Test",
        },
      },
    });

    await component.trigger("click");
    expect(component.emitted().ageClick).toBeTruthy();
  });
});
