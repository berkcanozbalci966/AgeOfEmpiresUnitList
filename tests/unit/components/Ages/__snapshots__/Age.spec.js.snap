// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age Render correctly 1`] = `<div class="" data-test="age"><span>Test</span></div>`;
